nixos/modules/system/boot/loader/raspberrypi/uboot-builder.sh

```diff
@@ -21,18 +22,35 @@ copyForced() {
 
 # Add the firmware files
 fwdir=@firmware@/share/raspberrypi/boot/
+copyForced $fwdir/overlays  $target/overlays
 copyForced $fwdir/bootcode.bin  $target/bootcode.bin
 copyForced $fwdir/fixup.dat     $target/fixup.dat
+copyForced $fwdir/fixup4.dat    $target/fixup4.dat
+copyForced $fwdir/fixup4cd.dat  $target/fixup4cd.dat
+copyForced $fwdir/fixup4db.dat  $target/fixup4db.dat
+copyForced $fwdir/fixup4x.dat   $target/fixup4x.dat
 copyForced $fwdir/fixup_cd.dat  $target/fixup_cd.dat
 copyForced $fwdir/fixup_db.dat  $target/fixup_db.dat
 copyForced $fwdir/fixup_x.dat   $target/fixup_x.dat
 copyForced $fwdir/start.elf     $target/start.elf
+copyForced $fwdir/start4.elf    $target/start4.elf
+copyForced $fwdir/start4cd.elf  $target/start4cd.elf
+copyForced $fwdir/start4db.elf  $target/start4db.elf
+copyForced $fwdir/start4x.elf   $target/start4x.elf
 copyForced $fwdir/start_cd.elf  $target/start_cd.elf
 copyForced $fwdir/start_db.elf  $target/start_db.elf
 copyForced $fwdir/start_x.elf   $target/start_x.elf
 
+# Add the config.txt
+copyForced @configTxt@ $target/config.txt
+
 # Add the uboot file
 copyForced @uboot@/u-boot.bin $target/u-boot-rpi.bin
 
-# Add the config.txt
-copyForced @configTxt@ $target/config.txt
+# Add the raspberry pi 4 specific files
+if [[ "@version@" == "4" ]]; then
```

Comment:
this was previously wrong. I noticed since I wiped my rpi4 /boot and these files were missing afterward.

Reply:
I wonder if we shouldn't rather support all aarch64 Raspberry Pi at once.

Right now configuring for the Raspberry Pi 4 means you *also* get the files for the Raspberry Pi 3 family, but not the inverse. Unless I'm mistaken?

Doing so would require `config.txt` to be aware of [the `[pi_]` sections](https://github.com/NixOS/nixpkgs/blob/e62feb3bf4a603e26755238303bda0c24651e155/nixos/modules/installer/sd-card/sd-image-aarch64.nix#L31-L37) I guess.

Reply:
I think your analysis is right, *and* the improvement you suggest would be the next/last big thing on the road to re-using this exact logic/builder for the sd-card image...

... but I can't sign-up for that right now.